splink/athena/athena_linker.py

```diff
@@ -344,9 +344,14 @@ def register_table(self, input, table_name, overwrite=False):
         self.register_data_on_s3(input, table_name)
         return self._table_to_splink_dataframe(table_name, table_name)
 
-    def _random_sample_sql(self, proportion, sample_size):
+    def _random_sample_sql(self, proportion, sample_size, seed=None):
         if proportion == 1.0:
             return ""
+        if seed:
+            raise NotImplementedError(
+                "Athena does not support seeds in random ",
+                "samples. Please remove the `seed` parameter.",
+            )
```

Comment:
Do we want this to fail or warn the user that no seed will be set and that u training will proceed?

Reply:
My first thought is for it to fail. Given we aren't setting the seed as a default and people's tendency to ignore warnings - but happy to change if you think it is better to throw a warning instead

Reply:
Given that the fail is virtually at the start of the process, I'd say it makes sense to leave it as a `NotImplementedError`